src/gsi/genstats_gps.f90

```diff
@@ -766,27 +768,33 @@ subroutine contents_netcdf_diag_
-           call nc_diag_metadata("Setup_QC_Mark",                         sngl(gps_allptr%rdiag(10))   )
-           call nc_diag_metadata("Prep_Use_Flag",                         sngl(gps_allptr%rdiag(11))   )
-           call nc_diag_metadata("Analysis_Use_Flag",                     sngl(gps_allptr%rdiag(12))   )
-
-           call nc_diag_metadata("Nonlinear_QC_Rel_Wgt",                  sngl(gps_allptr%rdiag(13))   )
-           call nc_diag_metadata("Errinv_Input",                          sngl(gps_allptr%rdiag(14))   )
-           call nc_diag_metadata("Errinv_Adjust",                         sngl(gps_allptr%rdiag(15))   )
-           call nc_diag_metadata("Errinv_Final",                          sngl(gps_allptr%rdiag(16))   )
-           call nc_diag_metadata("Observation",                           sngl(gps_allptr%rdiag(17))   )
-           call nc_diag_metadata("Obs_Minus_Forecast_adjusted",           sngl(gps_allptr%rdiag(17))*sngl(gps_allptr%rdiag(5)) )
-           call nc_diag_metadata("Obs_Minus_Forecast_unadjusted",         sngl(gps_allptr%rdiag(17))*sngl(gps_allptr%rdiag(5)) )
-           call nc_diag_metadata("GPS_Type",                              sngl(gps_allptr%rdiag(20))   )
-           call nc_diag_metadata("Temperature_at_Obs_Location",           sngl(gps_allptr%rdiag(18))   )
-           call nc_diag_metadata("Specific_Humidity_at_Obs_Location",     sngl(gps_allptr%rdiag(21))   )
+           call screen_to_single_nc_diag_metadata("Latitude",(gps_allptr%rdiag(3))    )
+           call screen_to_single_nc_diag_metadata("Longitude",(gps_allptr%rdiag(4))    )
+           call screen_to_single_nc_diag_metadata("Incremental_Bending_Angle",(gps_allptr%rdiag(5))    )
+           call screen_to_single_nc_diag_metadata("Pressure",(gps_allptr%rdiag(6))    )
+           call screen_to_single_nc_diag_metadata("Height",(gps_allptr%rdiag(7))    )
+           call screen_to_single_nc_diag_metadata("Time",(gps_allptr%rdiag(8))    )
+           call screen_to_single_nc_diag_metadata("Model_Elevation",(gps_allptr%rdiag(9))    )
+           call screen_to_single_nc_diag_metadata("Setup_QC_Mark",(gps_allptr%rdiag(10))   )
+           call screen_to_single_nc_diag_metadata("Prep_Use_Flag",(gps_allptr%rdiag(11))   )
+           call screen_to_single_nc_diag_metadata("Analysis_Use_Flag",(gps_allptr%rdiag(12))   )
+
+           call screen_to_single_nc_diag_metadata("Nonlinear_QC_Rel_Wgt",(gps_allptr%rdiag(13))   )
+           call screen_to_single_nc_diag_metadata("Errinv_Input",(gps_allptr%rdiag(14))   )
+           call screen_to_single_nc_diag_metadata("Errinv_Adjust",(gps_allptr%rdiag(15))   )
+           call screen_to_single_nc_diag_metadata("Errinv_Final",(gps_allptr%rdiag(16))   )
+           call screen_to_single_nc_diag_metadata("Observation",(gps_allptr%rdiag(17))   )
+           !TODO Verify that 'adjusted' is supposed to be the same as unadjusted
```

Comment:
Who is supposed to `TODO` this?

Reply:
Agreed that it is separate to this issue but the `TODO` comment will most likely be forgotten.   We can keep the comment but should also open a NOAA-EMC/GSI issue to publicize the fact that follow up is needed.

Reply:
Agreed.  I've opened #573 to address this.  The TODO can be removed.

Reply:
Thanks!